common/mempool.go

```diff
@@ -0,0 +1,32 @@
+package common
+
+type MemPool struct {
+	BlockSize int
+	PoolSize  int
+	memCh     chan []byte
+}
+
+func NewMemPool(blockSize, poolSize int) *MemPool {
+	return &MemPool{
+		BlockSize: blockSize,
+		PoolSize:  poolSize,
+		memCh:     make(chan []byte, poolSize),
+	}
+}
+
+func (pool *MemPool) Get() []byte {
+	var block []byte
+	select {
+	case block = <-pool.memCh:
+	default:
```

Comment:
Comment: If can't get the requested pool from MemPool, it will run the default branch which will generate a new pool and return it .